System/Posix/Files/Common.hsc

```diff
@@ -520,7 +520,7 @@ foreign import ccall unsafe "fchown"
 -- Note: calls @ftruncate@.
 setFdSize :: Fd -> FileOffset -> IO ()
 setFdSize (Fd fd) off =
-  throwErrnoIfMinus1_ "setFdSize" (c_ftruncate fd off)
+  throwErrnoIfMinus1_ "setFdSize" (c_ftruncate fd (fromIntegral off))
```

Comment:
why does this impedance mismatch occur? I'm reluctant to use a non-statically checked `fromIntegral` conversion which have been a common source of fileoffset overflow bugs... :-/